Source/Parallelization/WarpXComm_K.H

```diff
@@ -133,38 +133,18 @@ void warpx_interp (int j, int k, int l,
 
     // Refinement ratio
     const int rj = rr[0];
-#if defined(WARPX_DIM_1D_Z)
-    constexpr int rk = 1;
-    constexpr int rl = 1;
-#elif defined(WARPX_DIM_XZ) || defined(WARPX_DIM_RZ)
-    const     int rk = rr[1];
-    constexpr int rl = 1;
-#else
-    const int rk = rr[1];
-    const int rl = rr[2];
-#endif
+    const int rk = (AMREX_SPACEDIM > 1) ? rr[1] : 1;
+    const int rl = (AMREX_SPACEDIM > 2) ? rr[2] : 1;
 
     // Staggering of fine array (0: cell-centered; 1: nodal)
     const int sj_fp = arr_fine_stag[0];
-#if defined(WARPX_DIM_XZ) || defined(WARPX_DIM_RZ)
-    const int sk_fp = arr_fine_stag[1];
-#elif defined(WARPX_DIM_3D)
-    const int sk_fp = arr_fine_stag[1];
-    const int sl_fp = arr_fine_stag[2];
-#endif
+    const int sk_fp = (AMREX_SPACEDIM > 1) ? arr_fine_stag[1] : 1;
+    const int sl_fp = (AMREX_SPACEDIM > 2) ? arr_fine_stag[2] : 1;
 
     // Staggering of coarse array (0: cell-centered; 1: nodal)
     const int sj_cp = arr_coarse_stag[0];
-#if defined(WARPX_DIM_1D_Z)
-    constexpr int sk_cp = 0;
-    constexpr int sl_cp = 0;
-#elif defined(WARPX_DIM_XZ) || defined(WARPX_DIM_RZ)
-    const     int sk_cp = arr_coarse_stag[1];
-    constexpr int sl_cp = 0;
-#else
-    const int sk_cp = arr_coarse_stag[1];
-    const int sl_cp = arr_coarse_stag[2];
-#endif
+    const int sk_cp = (AMREX_SPACEDIM > 1) ? arr_coarse_stag[1] : 1;
+    const int sl_cp = (AMREX_SPACEDIM > 2) ? arr_coarse_stag[2] : 1;
```

Comment:
This seems to change the behavior of the code. For instance, if `WARPX_DIM_1D_Z` is defined `sk_cp` would be 1 instead of 0.  Is this intended?